src/psyclone/psyir/symbols/symbol_table.py

```diff
@@ -454,6 +460,15 @@ def find_or_create_tag(self, tag, root_name=None, **new_symbol_args):
         except KeyError:
             if not root_name:
                 root_name = tag
+            if exact_name:
+                symbols = self.get_symbols()
+                existing_names = set(symbols.keys())
+                if root_name in existing_names:
+                    raise SymbolError(
+                        f"Attempted to create a symbol with name "
+                        f"'{root_name}' but a symbol with that name already "
+                        f"exists, and using the exact name was required."
```

Comment:
Maybe "Can not create symbol '{root_name}' without renaming it because it already exists in this scope"